src/Common/src/System/Net/WebSockets/ManagedWebSocket.cs

```diff
@@ -279,7 +279,8 @@ public override Task SendAsync(ArraySegment<byte> buffer, WebSocketMessageType m
 
             try
             {
-               WebSocketValidate.ThrowIfInvalidState(_state, _disposed, s_validSendStates);
+                ThrowIfDisposed();
```

Comment:
If we wrap the ManagedWebSocket in HttpListener, and do the check in HttpListener instead (as it done in ClientWebSocket), for the ManagedWebSocket itself, it's still missing the functionality that to check for dispose before checking the invalid state. 

There could be a scenario that the code directly calls ManagedWebSocket methods instead of the methods in wrapper class (like ClientWebSocket or HttpListener). Therefore, I think add the check in ManagedWebSocket is better.